cmd/cli/namespace_add.go

```diff
@@ -142,3 +152,18 @@ func (a *namespaceAddCmd) run() error {
 
 	return nil
 }
+
+// getControllerDeployementsForMesh obtains all controller deployments within a mesh
+func getControllerDeployementsForMesh(clientSet kubernetes.Interface, meshName string, namespace string) ([]v1.Deployment, error) {
+	deploymentsClient := clientSet.AppsV1().Deployments(namespace)
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"meshName": meshName}}
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	}
+	osmControllerDeployments, err := deploymentsClient.List(context.TODO(), listOptions)
+	return osmControllerDeployments.Items, err
+}
+
+func errMeshNotFound(name string) error {
+	return errors.Errorf("Mesh [%s] does not exist. Please specify another mesh using --mesh-name or create a new mesh.", name)
+}
```

Comment:
why not `errors.Errorf()` directly instead of this intermediary function?

Reply:
Addressed in latest changes.